MobiFlight/Joysticks/Octavi/Octavi.cs

```diff
@@ -20,20 +20,35 @@ public Octavi(SharpDX.DirectInput.Joystick joystick, JoystickDefinition definiti
 
         }
 
+        /// <summary>
+        /// Method is not called by regular Joystick Manager
+        /// This method is called implicitly when Update() is called
+        /// </summary>
         public void Connect()
         {
-            if (Device == null)
+            // Prevent reentry and parallel execution by multiple threads
+            lock (this)
```

Comment:
Usually lock(this) is not recommended. Best practice is to use a separate dedicated new private object for a lock. Because it is out of control if someone else might use that object reference also for some locking.